src/ports/postgres/modules/utilities/path.py_in

```diff
@@ -118,140 +120,172 @@ def path(schema_madlib, source_table, output_table, partition_expr,
+        #       This is performed only for partitions that contain a match.
+        match_id_name = "__madlib_path_match_id__" if "match_id" in all_input_cols else "match_id"
+        symbol_name = "__madlib_path_symbol__" if "symbol" in all_input_cols else "symbol"
         plpy.execute("""
             CREATE TEMP TABLE {matched_partitions} AS
                 SELECT
                     {p_col_name_str},
-                    array_to_string(array_agg({symbol_name_str} ORDER BY {order_expr}), '') as sym_str,
-                    array_agg({id_col_name} ORDER BY {order_expr}) as matched_ids
+                    array_to_string(array_agg({short_sym_name_str} ORDER BY {order_expr}), '') as sym_str,
+                    array_agg({id_col_name} ORDER BY {order_expr}) as {match_to_row_id}
                 FROM {input_with_id}
+                WHERE {short_sym_name_str} is NOT NULL
                 GROUP BY {p_col_name_str}
-                HAVING array_to_string(array_agg({symbol_name_str} ORDER BY {order_expr}), '') ~* '{pattern_expr}'
+                HAVING array_to_string(array_agg({short_sym_name_str} ORDER BY {order_expr}), '')
+                                ~* '{new_pattern_expr}'
+            """.format(**locals()))
+
+        # length_of_matches: For each partition in matched_partitions:
+        #       - find all matches and compute the lengths of each match.
+        #       - output these lengths in an array (matches), along with
+        #         an array of corresponding rank of each match (match_indices).
+        length_of_matches = unique_string("match_length_view")
+        plpy.execute("""
+            CREATE VIEW {length_of_matches} AS
+            SELECT
+                {p_col_name_str},
+                {match_to_row_id},
+                array_agg(length(matches) ORDER BY match_index) AS matches,
+                array_agg(match_index ORDER BY match_index) AS match_indices
+            FROM (
+                SELECT
+                    {p_col_name_str},
+                    {match_to_row_id},
+                    generate_series(1, num_matches) AS match_index,
+                    (regexp_matches(
+                           sym_str, '(?i)({new_pattern_expr})', 'g'))[1] AS matches
+                FROM (
+                    SELECT
+                        {p_col_name_str},
+                        {match_to_row_id},
+                        sym_str,
+                        count(matches) AS num_matches
+                    FROM (
+                        SELECT
+                            {p_col_name_str},
+                            {match_to_row_id},
+                            sym_str,
+                            (regexp_matches(
+                                   sym_str, '(?i)({new_pattern_expr})', 'g'))[1] AS matches
+                        FROM
+                           {matched_partitions}
+                        ) t1
+                    GROUP BY {p_col_name_str}, {match_to_row_id}, sym_str
```

Comment:
no it can't. the count(...) is the only aggregate there and we need the other columns in subsequent outer query.